lcls_naming_tool/tests/test_cli.py

```diff
@@ -0,0 +1,86 @@
+# Unit tests for the LCLS Naming Tool script
+
+import pytest
+import json
+from pathlib import Path
+from lcls_naming_tool import display_version, load_taxons, functional_component_is_valid, fungible_is_valid, constituent_component_is_valid, increment_is_valid, starts_alphanumeric, validate
+
+
+@pytest.fixture
+def valid_name():
+    return ['MR2K4', 'KBO', 'PIP', '01', 'PUMPSIZE']
+
+
+@pytest.fixture
+def invalid_name():
+    return ['ABC12345M9', 'ABC', 'NBC', '7', '_PCTRLSPRBCK']
+
+
+def test_display_version(capsys):
+
+    display_version()
+    captured = capsys.readouterr()
+    assert str(captured.out).strip() == 'v1.0.0'
+
+
+load_taxons()
+
+
+def test_functional_component_is_valid_true(valid_name):
```

Comment:
Pytest lets you parametrize these tests so you don't have to copy paste.  This also lets you expand the test matrix more easily as we find and want to test more edge cases.  See: https://docs.pytest.org/en/stable/how-to/parametrize.html
```python
VALID_NAMES = ['MR2K4', 'KBO', 'PIP', '01', 'PUMPSIZE']


@pytest.mark.parametrize('name', VALID_NAMES)
def test_valid_name(name: str):
    assert functional_component_is_valid(name)
    assert fungible_is_valid(name)
```

And similarly for invalid names etc

Reply:
Hmm when I do this the `assert fungible_is_valid(name)` seems to come back with an error.
```
E       AssertionError: assert False == True
E        +  where False = fungible_is_valid('MR2K4')
```

Reply:
If you hit an `AssertionError`, that means your test is failing!

Reply:
Ok, think I figured out how @pytest.mark.parametrize works. Updated the tests.

Reply:
If in doubt, you can always look through `pcdshub` for examples.  We use pytest rather extensively